### ISSUE TITLE
[Feature Request] Link to other notes, Wiki-Style

### ISSUE THREAD

**Issue by ionisingTuna** [standardnotes/forum#324]
I think a feature to create links to other notes would be very beneficial.

**Comment by mobitar** [comment]
Hi @ionisingTuna, some related topics here:

https://github.com/standardnotes/forum/issues/43
https://github.com/standardnotes/desktop/issues/181
https://github.com/standardnotes/forum/issues/293
https://github.com/standardnotes/forum/issues/251
https://github.com/standardnotes/web/issues/126

**Comment by lvvvvvl** [comment]
+1 this. I love everything about StandardNotes but without the internal linking feature, I can't commit to using it. 

To me the 'longevity' philosophy not only means that I shouldn't feel locked-in by the features like I do with Evernote, but that certain features contribute to the growth of my Notes database creating a positive feedback loop; the longer I use my notes and keep them organized by interlinking them, the more value my notes will have for me and the longer I'll use them.

**Comment by justinthrelkeld** [comment]
+1 on linking to other notes. I tried to make the switch over to Standard Notes but this feature is critical. Reading the discussions tagged above makes it difficult to get a sense for whether this is something the project will ever consider: sometimes it sounds like a long-term priority that will definitely (or maybe? or possibly?) happen eventually but other times it sounds like it's considered out of scope.

I would love to contribute to and support the project, but without any cross-note linking I can't even begin using Standard Notes. If this feature is on the roadmap, I would love to help make it happen.

**Comment by mobitar** [comment]
I think it will eventually make it into SN, but there are a few core things needing to be done ahead of it. I can't or won't give definitive statements because that can turn into a relentless attack of "are we there yet?!" from users ;) Not to mention how volatile software roadmaps can be.

**Comment by DavBE** [comment]
+1 would love this

**Comment by mpern** [comment]
This could be something like `note://<uid>`

**Comment by ajvsol** [comment]
I'd like it to be like a URL, so I can have a bookmark or a hyperlink from different software I use which takes me directly to the opened note.

**Comment by russlipton** [comment]
+1. Again. Lack of this feature keeps me from using SN seriously and, especially, Listed.

**Comment by Cito** [comment]
Internal links and notifications are my most wanted features, too.

**Comment by annargrs** [comment]
Same here, I really wanted to give SN a try but this is a non-starter.

**Comment by guydog** [comment]
i second this feature request.

This is a crucial one for making SN  a consistent working tool and not "just" a notepad. 

Actually for my work I am considering switching back to zim, gnote/tomboy or similar (in Linux) precisely because I need this feature: my "corpus" of notes in SN is growing up and I need to organize it at a different level. But this way I would loose all the benefits of SN...

Please put the feature on the top side of your agenda!
(could the code be "borrowed" from already existing software?)

Thank you
G.

**Comment by the-V** [comment]
Yeah, just adding to the sentiment.
This was all that was missing for me to pull the trigger on Standard Notes and shell out for the 5 year subscription. Not being able to make a personal wiki or and index defeats the purpose of a notes app for me.

**Comment by hbri** [comment]
Just purchased a subscription. So many great things about this app, but note linking is on my short list of features I’d really like to see added. Any news on this?

**Comment by genejo** [comment]
I also second this request. Without this feature it's just a matter of time before a nice set of notes becomes a heavy bag of unrelated (yet relatable) pieces of information.

If notes are to be safely stored for 100 years, they should also be easily accessible after that time.

@annargrs Though it's very different from SN, as a markdown editor NotePlan implements this feature elegantly, but the software just for the apple ecosystem.

**Comment by plantfansam** [comment]
Cosign! Would love this!

**Comment by Zie0** [comment]
I may have to consider the $150 I paid for standard notes a donation while I wait for this feature to be implemented. I get the longevity idea and that was a big part of the reason I got excited about SN! The lack of [this feature](https://standardnotes.org/help/58/how-do-i-link-multiple-notes-together-to-create-a-sort-of-personal-wiki) is on the border between dealbreaker and killer app. 

For all of Joplin's shortcomings I may decide to keep using it instead

**Comment by thed0ctor** [comment]
+1 for this feature. I'm coming from Bear where this feature was used extensively

**Comment by Cito** [comment]
Currently people start to recognize the potential of note-taking combined with Wiki-style linking, and freak out if it's even bi-directional. Whole [cults](https://goodbetterbest.substack.com/p/roam-research-the-cult-packaging) have already formed around this idea. I think Standard Notes should ride that wave, and it would still stand out among the emerging tools like Roam, Obsidian, Logseq, Athens, Notebag, Amplenode or RemNote because of its outstanding encryption and syncing capabilities.

**Comment by mobitar** [comment]
Linking is on our minds for v4. Exactly what form it will take will be tbd.

**Comment by VegasVed** [comment]
Cool. What's the ETA on v4?

**Comment by mobitar** [comment]
We don't ask that question ;)

**Comment by anavarre** [comment]
You need a bot that returns https://blog.standardnotes.org/being-a-quiet-software-company/ when this question is being asked.

**Comment by Zie0** [comment]
oops! I realize that this may be a request for a core free version feature. 

is it possible this could be an Extension? I would happily pay for it. are there blockers that keep this from being an extension?

**Comment by ghost** [comment]
I would also happily pay for the feature

**Comment by the-V** [comment]
I would also happily pay for the feature

Get Outlook for Android<https://aka.ms/ghei36>

________________________________

**Comment by justinthrelkeld** [comment]
Likewise: even with the explosion of other note-taking applications, I am still very interested in Standard Notes for a number of reasons (the quite company aspect among them) and I would happily pay for the ability to create links

**Comment by theontho** [comment]
I would recommend doing this roam research style, where `#mytag` or `[[note name]]` would still be very 'long term easy storage' style and would integrate well with general markdown editing as an extension.  Then in the future you can add backlinks like roam and https://www.outline.md/ does.

**Comment by Cito** [comment]
@theontho This would also be compatible with Obsidian.

**Comment by JaspalSuri** [comment]
For anyone who doesn't mind using a CLI, they could use what [this user suggested on reddit](https://www.reddit.com/r/StandardNotes/comments/is2r7e/a_way_to_use_bidirectional_linking_with_your_sn/) on the desktop (and perhaps even some Android devices).

**Comment by Cito** [comment]
@JaspalSuri That way you could also use a good external Markdown editor like Typora on your Standard Notes. However, the problem is when you forget to check in and out propery, and work in Standard Notes and on the filesystem at the same time in parallel, you can easily screw things up.

**Comment by virtuallyvlad** [comment]
> For anyone who doesn't mind using a CLI, they could use what [this user suggested on reddit](https://www.reddit.com/r/StandardNotes/comments/is2r7e/a_way_to_use_bidirectional_linking_with_your_sn/) on the desktop (and perhaps even some Android devices).

Thanks for linking to this.  I also saw this approach mentioned in the [Fast Editor](https://github.com/dec0dOS/standard-notes-fast-editor) readme but am confused as to what the workflow would be like if using their editor.  Fast Editor lets you have a shortcut in SN that adds a markdown link to the note that is basically just the timestamp (like `[[202106191655]]`), but since it doesn't actually work as a link I'm guessing you'd have to then go back to VS Code to actually use the link?  So I'm not sure what would be the point of having this shortcut in an SN editor if you have to switch to another program to then be able to use the links.

Also, [standardnotes-fs](https://github.com/tannercollin/standardnotes-fs) only works with protocol version 003 so it seems like this will no longer be viable for anyone who's upgraded to 004+.

**Comment by alat-rights** [comment]
I'd like to take a stab at this. Would it be a good idea for me to start working on a PR, or would be that nonviable (either because of existing internal headway or complexities I don't grok)?

**Comment by mobitar** [comment]
I don't think this would be the best candidate for external contribution due to the complexities involved, especially the cross-platform nature of it. But if you'd like to work on it to serve as an informational guide for us on how potential solutions could look like, by all means!

**Comment by alat-rights** [comment]
Sounds good! I'll have to familiarize myself a little bit more with the code base first.

**Comment by chriscz** [comment]
I've not used Roam, but I see they also offer a linking mechanism, would be interesting to see what they did and how/if some of it could be used within standard notes: https://roamresearch.com/

**Comment by nvkv** [comment]
In the spirit of "whistles is someone else's problem" it would be enough to get notes *addressable* in any way. By ID, by name, by unique tag it doesn't really matter. You don't even need to implement linking support in all editors, it seems you guys weren't worried much about FileSafe links anyway as they doesn't work anywhere except Bold Editor ;)

Just give us foundation, we'll pick it up from here.

**Comment by orionp** [comment]
Perhaps another approach would be worth considering? Though it may have already been suggested, I've not yet found it in the forum, so I'll add it to this conversation.

[The Archive](https://zettelkasten.de/the-archive/) has a pretty slick way of doing links, which may adapt to the Standard Notes approach more easily than some other approaches.

Rather than a link being a link to a specific file, it's a link to a search term. If you are using their date/time Zettelkasten timestamp (e.g. `202111112159`), then a link for that timestamp format returns a unique note (or notes containing links to that timestamp), but I've found it incredibly useful for quick linking, as well. `[[2021-11-11]]` will return all notes from today, while `[[Planning Meeting 2021-11-11]]` will return the meeting from today, as well as any other notes linking to that meeting.

This also allows some flexibility in linking from the outside world; rather than needing a definitive link to a note in an encrypted database, you can search for a given string using e.g. `thearchive://search/TERM` or `thearchive://match/TERM`.

More details:

- [How do I create links between notes?](https://zettelkasten.de/the-archive/help/#how-do-i-create-links-between-notes)
- [How do I link to notes from other apps?](https://zettelkasten.de/the-archive/help/#how-do-i-link-to-notes-from-other-apps)

**Comment by viharsheth** [comment]
I appreciate the workarounds being suggested but note linking / back linking seems like a must add feature for SN to remain relevant. I've been waffling between Obsidian and SN for some time now. Obsidian is near perfect expect there is no web app. If it had one I would have even abandoned SN by now, even though I'm paid up through 2034! If SN could add note linking (and native image support) it would be _perfect_!

**Comment by DJTimoy** [comment]
Backlinking is a "simple" function that adds enormous capability with organizing notes. They are so incredibly useful with assimilating one's note-taking environment with the way our brains function. It is now an essential feature for note making, along with #tags and search.

Cross-Note Links are basically one-way links while Backlinks / Bi-Directional links provide a way back too.

**Comment by atmoio** [comment]
It's on our radar. Native encrypted files/images is our main big picture priority right now. Linking will likely be after that.

**Comment by effieeee** [comment]
It's been out for a little while, but it's now possible to link notes to one another!
![image](https://user-images.githubusercontent.com/54989748/198839610-e9a8c0b9-b4af-4edd-b6a7-115f9f856ab6.png)

I understand that it might not be the exact means of note linking that others might have wanted though, so feel free to continue the discussion here 😅

**Comment by thomaseizinger** [comment]
> It's been out for a little while, but it's now possible to link notes to one another! ![image](https://user-images.githubusercontent.com/54989748/198839610-e9a8c0b9-b4af-4edd-b6a7-115f9f856ab6.png)

I love it, I think it is a great idea! Also to have files show up there is quite neat!

**Comment by Cito** [comment]
@effieeee Since which version is this possible and can you explain a bit more how it works?

**Comment by effieeee** [comment]
Hi @Cito, I believe it was launched sometime around v3.23.200 onwards! From then on, what used to be the tag entry box under the note title, has been turned into an all-in-one **item linking** space, where after typing a particular keyword, the dropdown menu will display suggestions of tags, files, and notes that you'd like to link to the current note.

When you choose to link a **note**, this creates a **bidirectional** link towards your chosen note, such that when you click on that note's title in the item linking space, you can immediately switch to that note, and vice versa. Alternatively, you can also click on the **linking icon** at the top right of the app to view a more organized list of the currently linked tags, files, and notes. You may choose to link items here instead too!

![image](https://user-images.githubusercontent.com/54989748/200347229-889fd4c0-385d-4074-ac86-d60a7d49a70c.png)

**Comment by Cito** [comment]
Hi @effieeee thanks for the explanation. I hadn't expected such a big feture would be added in a patch level release, but after installing the latest patch level it is working for me. too.

This is not really wiki-style linking, but close enough - and maybe fits better into the overall paradigm how Standard Notes works. Nice!

**Comment by DJTimoy** [comment]
@effieeee woaahh awesome! this is epic, I don't think any notes app is doing this but it makes sooo much sense.
So much less clicking around, wondering "where's that file??" when everything can be centralized and interconnected **very** appropriately.

The only thing that would make this "complete" per-say, is if we could somehow _**"display attachments inline"**_.

(I think, Obsidian uses markdown links to insert attachments, so perhaps something can be automated with how your attachments work)
Then we can get that "wiki-style" linking and so much more, it would blow everything out of the water. There would be no reason to use Notion or something if Standard Notes is objectively better in every way!

**Comment by DJTimoy** [comment]
Pardon me while I check out the "Super Editor" 🤔

Apparently, you have indeed accomplished wiki-style linking! This is absolutely massive..

**Comment by thomaseizinger** [comment]
> Pardon me while I check out the "Super Editor" 🤔
> 
> Apparently, you have indeed accomplished wiki-style linking! This is absolutely massive..

Thank you for posting this, I had no idea about this "Super" editor!

Is there a blog / mailing list / something else I can subscribe to where the team writes about what they are building? I'd love to read about that!
